quick_start_demo/5-min-Mosaic-AI-Agents-Demo-with-Synthetic-Eval.py

```diff
@@ -0,0 +1,699 @@
+num_evals = 25
+evals = generate_evals_df(
+    docs=parsed_docs_df[:500], # Pass your docs. Pandas/Spark Dataframes with columns `content STRING, doc_uri STRING` are suitable.
+    num_evals=num_evals, # How many synthetic evaluations to generate
+    guidelines=guidelines
+)
+display(evals)
+
+# COMMAND ----------
+
+# MAGIC %md
+# MAGIC # 2/ Write the Agent's code
+
+# COMMAND ----------
+
+# MAGIC %md
+# MAGIC ## 2.1/ Function-calling agent w/ a Retriever Tool
+
+# COMMAND ----------
+
+# MAGIC %md
+# MAGIC **Challenges addressed**
+# MAGIC - How do I track different versions of my agent's code/config?
+# MAGIC - How do I enable observability, monitoring, and debugging of my Agent’s logic?
+# MAGIC
+# MAGIC **What is happening?**
+# MAGIC
+# MAGIC First, we create a function-calling Agent with access to a retriever tool using OpenAI SDK + Python code.  To keep the demo simple, the retriever is a function that performs keyword lookup rather than a vector search index.
+# MAGIC
+# MAGIC When creating your Agent, you can either:
+# MAGIC 1. Generate template Agent code from the AI Playground
+# MAGIC 2. Use a template from our Cookbook
+# MAGIC 3. Start from an example in popular frameworks such as LangGraph, AutoGen, LlamaIndex, and others.
+# MAGIC
+# MAGIC **NOTE: It is not necessary to understand how this Agent works to understand the rest of this demo notebook.**  
+# MAGIC
+# MAGIC *A few things to note about the code:*
+# MAGIC 1. The code is written to `fc_agent.py` in order to use [MLflow Models from Code](https://www.mlflow.org/blog/models_from_code) for logging, enabling easy tracking of each iteration as we tune the Agent for quality.
+# MAGIC 2. The code is parameterized with an [MLflow Model Configuration](https://docs.databricks.com/en/generative-ai/agent-framework/create-agent.html#use-parameters-to-configure-the-agent), enabling easy tuning of these parameters for quality improvement.
+# MAGIC 3. The code is wrapped in an MLflow PyFunc model, making the Agent's code deployment-ready so any iteration can be shared with stakeholders for testing.
+# MAGIC 4. The code implements [MLflow Tracing](https://docs.databricks.com/en/mlflow/mlflow-tracing.html) for unified observability during development and production. The same trace defined here will be logged for every production request post-deployment. For agent authoring frameworks like LangChain and LlamaIndex, you can tracing with one line of code: `mlflow.langchain.autolog()`/`mlflow.llama_index.autolog()`.
+
+# COMMAND ----------
+
+# MAGIC %%writefile fc_agent.py
+# MAGIC from databricks.sdk import WorkspaceClient
+# MAGIC from openai import OpenAI
+# MAGIC import pandas as pd
+# MAGIC from typing import Any, Union, Dict, List
+# MAGIC import mlflow
+# MAGIC from mlflow.models.rag_signatures import ChatCompletionResponse, ChatCompletionRequest
+# MAGIC import dataclasses
+# MAGIC import json
+# MAGIC
+# MAGIC DEFAULT_CONFIG = {
+# MAGIC     'endpoint_name': "agents-demo-gpt4o",
```

Comment:
let's change this to FMAPI llama 70b

Reply:
we need to re-think about the LLM endpoint choices in this notebook.
I have changed the default config in `fc_agent.py` and also the `dev_config` of the vibe check section. However, the baseline config still has `"endpoint_name": "agents-demo-gpt4o""agents-demo-gpt4o"`. 
More importantly, it makes me realize that the notebook requires the user have 2 external model endpoints (`agents-demo-gpt4o` and `agents-demo-gpt4o-mini`) to succeed the "Run All". That should be mentioned in the requirements and 0/setup. I think it is better to remove the external model dependencies, however, all the screenshots show gpt models.